packages/components/src/ResourcePicker/Resource/Resource.component.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { distanceInWordsToNow } from 'date-fns';
+import { translate } from 'react-i18next';
+import I18N_DOMAIN_COMPONENTS from '../../constants';
+import getDefaultT from '../../translate';
+import getLocale from '../../DateFnsLocale/locale';
+import Icon from '../../Icon';
+
+import theme from './Resource.scss';
+
+function getDateLabel(t, date) {
+	return distanceInWordsToNow(date, {
+		addSuffix: true,
+		locale: getLocale(t),
+	});
+}
+
+function getAuthorLabel(t, author, date) {
+	return t('RESOURCE_OWNED_BY', {
+		defaultValue: 'owned by {{ author }}, {{date}}',
+		author,
+		date: getDateLabel(t, date),
+	});
+}
+
+function Resource({ parent, index, style, t }) {
+	const item = parent.props.collection[index];
```

Comment:
Can you rely on `components/src/VirtualizedList/utils/gridrow.js` > `getRowData()` instead ?
This will make proptypes simpler too.